servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ToStH1Utils.java

```diff
@@ -151,19 +151,35 @@ static Http2Headers h1HeadersToH2Headers(HttpHeaders h1Headers) {
         Iterator<? extends CharSequence> teItr = h1Headers.valuesIterator(TE);
         boolean addTrailers = false;
         while (teItr.hasNext()) {
-            String teValue = teItr.next().toString();
-            int i = teValue.indexOf(',');
-            if (i != -1) {
-                int start = 0;
-                do {
-                    if (teValue.substring(start, i).compareToIgnoreCase(TRAILERS.toString()) == 0) {
+            final CharSequence teSequence = teItr.next();
+            if (addTrailers) {
+                teItr.remove();
+            } else {
+                final String teValue = teSequence.toString();
```

Comment:
Looks like we don't need conversion to `String`, consider using `CharSequences.indexOf` and then `CharSequence.subSequence` with `contentEqualsIgnoreCase`